app/managers/websocket_manager.ts

```diff
@@ -227,35 +231,40 @@ class WebsocketManager {
         }
 
         const isActive = appState === 'active';
-        this.handleStateChange(this.netConnected, isActive);
+        this.handleStateChange(this.netConnected, this.netType, isActive);
     };
 
     private onNetStateChange = (netState: NetInfoState) => {
         const newState = Boolean(netState.isConnected);
-        if (this.netConnected === newState) {
-            return;
-        }
-
-        this.handleStateChange(newState, this.previousActiveState);
+        this.handleStateChange(newState, netState.type, this.previousActiveState);
     };
 
-    private handleStateChange = (currentIsConnected: boolean, currentIsActive: boolean) => {
-        if (currentIsActive === this.previousActiveState && currentIsConnected === this.netConnected) {
+    private handleStateChange = (currentIsConnected: boolean, currentNetType: NetInfoStateType, currentIsActive: boolean) => {
+        if (currentIsActive === this.previousActiveState && currentIsConnected === this.netConnected && currentNetType === this.netType) {
             return;
         }
 
         this.cancelConnectTimers();
 
         const wentBackground = this.previousActiveState && !currentIsActive;
+        const switchedNetworks = currentIsConnected && currentNetType !== this.netType && this.netType !== 'none';
 
         this.previousActiveState = currentIsActive;
         this.netConnected = currentIsConnected;
+        this.netType = currentNetType;
 
         if (!currentIsConnected) {
             this.closeAll();
             return;
         }
 
+        if (switchedNetworks) {
+            // Close all connections when we switch from (for example) vpn to wifi
+            // to ensure we are using the right network and doesn't get stuck on
+            // retries.
+            this.closeAll();
```

Comment:
You mean the network quickly moves between, let's say, wifi and cell?

I don't expect that to happen often, but it is something we could consider. @rahimrahman thoughts on this?

Reply:
0/5. I know you can toggle the Wifi switch back and forth quickly, but I doubt that it will get picked up by netinfo to worry about putting a debounce.